### ISSUE TITLE
Href links as values?

### ISSUE THREAD

**Issue by davecap** [plotly/dash-table-experiments#6]
Is it possible to embed links as cell values?

**Comment by chriddyp** [comment]
It isn't right now, but it would be nice to allow hooks into the cell renderer somehow.

**Comment by jamesdhope** [comment]
A great resource and it would be useful to develop the following:
- selectable columns (not just rows)
- ability to add new columns and rows dynamically to the table

More sample code would be really useful too with the plotly Library.

**Comment by zachspar** [comment]
Do you think this will ever be a feature of a DataTable? I'm running into the same issue of wanting to embed a link within the cell of a table. It would be super awesome!
 - If not, does anyone have any alternate ideas as to how I can go about embedding a link into a similarly formatted table?

**Comment by aiy2018** [comment]
I was able to use the following conditional formating idea to implement cell hyper link.
https://community.plot.ly/t/coloured-text-in-a-table-conditional-formatting/7112/6

Here is my method - if you have columns like "ABCDE", column A and B are hyper link
def ConditionalTable(dataframe):
    rows = []
    hyper_link_col_list = ['A', 'B']
    for i in range(len(dataframe)):
        row = {}
        for col in dataframe.columns:
            value = dataframe.iloc[i][col]
            if col in hyper_link_col_list:
                if "https" not in value:
                    value = "https://"+value
                row[col] = html.Td(html.A(href=value, children=value))
            else:
                row[col]=value
        rows.append(row)
    return rows

@app.callback(Output('table', 'rows'), [Input('field-dropdown', 'value')])
def update_table(user_selection):
    """
    For user selections, return the relevant table
    """
    df = get_data_object(user_selection)
    rows = ConditionalTable(df)
    df = pd.DataFrame(rows)
    return df.to_dict('records')

You need the following packages. 
 dash==0.19.0
 dash-core-components==0.16.0rc1
 dash-html-components==0.9.0rc1
 dash-renderer==0.13.0rc2
 dash-table-experiments==0.6.0rc1

For table sorting  and filtering on href links columns are not working since row[col] ={Td}Td('A') NOT row[col]= 'A'. This needs to be enhanced to be removed.
row[col] = html.Td(html.A(href=value, children=value))

**Comment by Madhivarman** [comment]
@aiy2018  How can we pass the Hyperlink value or **dcc.Link** component to another page?

**Comment by Madhivarman** [comment]
@aiy2018  I used the same method you did to insert Hyperlink in the cell. But dataTable doesn't shown up in the div element. This is the code I used,

```
def ConditionalTable(df):
	#Args(value): Type -> DataFrame
	rows = []
	hyperlink_cols = ['Unique_Id']

	for i in range(len(df)):
		row = {}
		for col in df.columns:
			value = df.iloc[i][col]
			#set an condition
			if col in hyperlink_cols:
				row[col]= html.Td(html.A(value, href='{}'.format(value)))
			else:
				row[col] = value

		rows.append(row)

	return rows

@app.callback(
	Output('snapshot_info', 'children'),
	[Input('intermediate-data', 'children')]
)

def display_the_datatable(df):
	data = json.loads(df)
	dataframe = pd.DataFrame.from_dict(data, orient='columns')

	rows = ConditionalTable(dataframe) #rows

	#convert into df again
	return html.Div([
		dt.DataTable(
			rows = rows,
			columns = sorted(dataframe.columns),
			filterable = True,
			editable=False,
			sortable = True,
			selected_row_indices= [],
			id= 'datatable-snapshot-Data'
		)
	], style={'display':'block'})
```
what's wrong? I checked if there is traceback error in the console but there is no error. How do I display the table in the div element.

**Comment by Madhivarman** [comment]
@aiy2018  Did you implemented the Link into the cell?

**Comment by aiy2018** [comment]
Yes. The above code is working for link in the cells except that the cell data cannot be sorted.
 I have not done passing the Hyperlink value to another page.
